include/envoy/network/listener.h

```diff
@@ -203,6 +208,18 @@ class TcpListenerCallbacks {
   virtual void onReject() PURE;
 };
 
+/**
+ * Callbacks invoked by a internal listener.
+ */
+class InternalListenerCallbacks {
+public:
+  virtual ~InternalListenerCallbacks() = default;
+
+  virtual void setupNewConnection(Network::ConnectionPtr server_conn,
```

Comment:
setUpNewConnection I think?

Also can you do javadoc style comments for these?